pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -502,6 +502,14 @@ func (c *Cluster) getOSDInfo(d *appsv1.Deployment) (OSDInfo, error) {
 		}
 	}
 
+	// if the ROOK_TOPOLOGY_AFFINITY env var was not found in the loop above, detect it from the node
+	if isPVC && osd.TopologyAffinity == "" {
+		osd.TopologyAffinity, err = getTopologyFromNode(c.context.Clientset, d, osd)
+		if err != nil {
+			logger.Errorf("failed to get topology affinity for osd %d", osd.ID)
```

Comment:
should we return the error rather than just logging it? (as the error getting `TopologyAffinity` might again result in the original issue)

Reply:
Good question. I was following the pattern of the next code block to look up the location. Since the OSDs have already been running and this is an upgrade, I believe we should treat this as a best effort. If it does fail, the next time a reconcile runs it will naturally try again anyway, but it does not feel like a fatal condition for updating the OSD pod spec. And this will only be run once per OSD for this one upgrade, so there is very little frequency to hit an error. After it is successful once for the OSD, the ROOK_TOPOLOGY_AFFINITY env var will be re-used in future reconciles.